earthnet_models_pytorch/utils/parse.py

```diff
@@ -82,32 +84,48 @@ def parse_setting(setting_file, track = None):
     setting_dict["Seed"] = setting_dict["Seed"] if "Seed" in setting_dict else 42
     setting_dict["Data"]["val_split_seed"] = setting_dict["Seed"]
 
-    setting_dict["Trainer"]["precision"] = setting_dict["Trainer"]["precision"] if "precision" in setting_dict["Trainer"] else 32
-    setting_dict["Data"]["fp16"] = (setting_dict["Trainer"]["precision"] == 16)
+    setting_dict["Trainer"]["precision"] = setting_dict["Trainer"]["precision"] if "precision" in setting_dict["Trainer"] else 32  # binary floating-point computer number format 
+    setting_dict["Data"]["fp16"] = (setting_dict["Trainer"]["precision"] == 16)   # binary floating-point computer number format 
 
     setting_dict["Checkpointer"] = {**setting_dict["Checkpointer"], **METRIC_CHECKPOINT_INFO[setting_dict["Setting"]]} if "Checkpointer" in setting_dict else METRIC_CHECKPOINT_INFO[setting_dict["Setting"]]
-
+    
     bs = setting_dict["Data"]["train_batch_size"]
     gpus = setting_dict["Trainer"]["gpus"]
-    ddp = (setting_dict["Trainer"]["accelerator"] == "ddp")
+    ddp = (setting_dict["Trainer"]["strategy"] == "ddp")
+    
     optimizers = setting_dict["Task"]["optimization"]["optimizer"]
-    for optimizer in optimizers:
+    for optimizer in optimizers:  
         if "lr_per_sample" in optimizer:
             lr_per_sample = optimizer["lr_per_sample"]
-            lr = bs * (gpus * ddp + (1-ddp)) * lr_per_sample
+            if isinstance(gpus, list):
+                lr = bs * (len(gpus) * ddp + (1-ddp)) * lr_per_sample
+            else:    
+                lr = bs * (gpus * ddp + (1-ddp)) * lr_per_sample
+            print('learning rate', lr)
             optimizer["args"]["lr"] = lr
         
-    if track is not None:
```

Comment:
please keep this part. The idea is to set the correct context and target lengths for a given setting. We can change this later as part of the refactoring.